vulcanai/models/metrics.py

```diff
@@ -1012,3 +1014,85 @@ def cross_validate(network, data_loader, k, epochs,
             logger.info(all_results)
             return all_results
 
+    def conduct_sensitivity_analysis(self, network, data_loader, filename,
+                                     features, cutoff=20):
+        """
+        Will conduct tests to figure out directionality of features by finding all the unique feature values present in
+        the dataset and setting the feature for all examples to every unique value of that feature.
+
+        Parameters
+        ----------
+            network : BaseNetwork
+                Network descendant of BaseNetwork.
+            data_loader : DataLoader
+                A DataLoader object which contains all the data
+            filename: string
+                The name of the file to save the test results to
+            cutoff: int
+                Maximum number of unique feature values for a particular feature that will be tested
+
+        Returns
+        -------
+        #TODO
+        """
+
+        if network._num_classes is None or network._num_classes == 0:
+            raise ValueError('There\'s no classification layer')
+
+        dataset = data_loader.dataset
+
+        test_df = None
+        col_headers = None
+
+        for feature in range(len(features)):
+            feature_val_list = dataset[:][0][:, feature].tolist()
+
+            unique_feature_values = list(np.unique(feature_val_list))
+
+            if len(unique_feature_values) > cutoff:
+                print("================================")
+                print("Cutting off number of features to be tested for feature "
+                      "{} from {} to {}".format(features[feature],
+                                                len(unique_feature_values),
+                                                cutoff))
+                print("================================")
+                unique_feature_values = unique_feature_values[0::int(
```

Comment:
what is this line doing?

Reply:
Since unique_feature_values are now sorted, it pulls cutoff # of features evenly spaced from the list

Reply:
I made a small mistake, will update in the next pr, it needs a math.ceil() instead of an int()

Reply:
ok let's add a detailed comment about that. it's not immediately obvious what's going on